src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -738,5 +716,312 @@ private static bool IsDynamicStatics(TypeDesc type)
+            }
+            else
+            // In ReadyToRun, we always use the dispatch stub to call virtual methods
+            {
+                pResult->kind = CORINFO_CALL_KIND.CORINFO_VIRTUALCALL_STUB;
+
+                if (pResult->exactContextNeedsRuntimeLookup)
+                {
+                    ComputeLookup(ref pResolvedToken,
+                        GetRuntimeDeterminedObjectForToken(ref pResolvedToken),
+                        ReadyToRunHelperId.VirtualDispatchCell,
+                        ref pResult->codePointerOrStubLookup);
+                    Debug.Assert(pResult->codePointerOrStubLookup.lookupKind.needsRuntimeLookup);
+                }
+                else
+                {
+                    pResult->codePointerOrStubLookup.lookupKind.needsRuntimeLookup = false;
+                    pResult->codePointerOrStubLookup.constLookup.accessType = InfoAccessType.IAT_PVALUE;
+                    pResult->codePointerOrStubLookup.constLookup.addr = (void*)ObjectToHandle(
+                        _compilation.SymbolNodeFactory.InterfaceDispatchCell(targetMethod,
+                            new ModuleToken(_tokenContext, (mdToken)pResolvedToken.token), _signatureContext, isUnboxingStub: false
+                        , _compilation.NameMangler.GetMangledMethodName(MethodBeingCompiled).ToString()));
+                }
+            }
+
+            pResult->hMethod = ObjectToHandle(targetMethod);
+
+            pResult->accessAllowed = CorInfoIsAccessAllowedResult.CORINFO_ACCESS_ALLOWED;
+
+            // We're pretty much done at this point.  Let's grab the rest of the information that the jit is going to
+            // need.
+            pResult->classFlags = getClassAttribsInternal(targetMethod.OwningType);
+
+            pResult->methodFlags = getMethodAttribsInternal(targetMethod);
+            Get_CORINFO_SIG_INFO(targetMethod, &pResult->sig, targetIsFatFunctionPointer);
+
+            if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_VERIFICATION) != 0)
+            {
+                if (pResult->hMethod != pResolvedToken.hMethod)
+                {
+                    pResult->verMethodFlags = getMethodAttribsInternal(targetMethod);
+                    Get_CORINFO_SIG_INFO(targetMethod, &pResult->verSig);
+                }
+                else
+                {
+                    pResult->verMethodFlags = pResult->methodFlags;
+                    pResult->verSig = pResult->sig;
+                }
+            }
+
+            pResult->_secureDelegateInvoke = 0;
+
+            switch (pResult->kind)
+            {
+                case CORINFO_CALL_KIND.CORINFO_VIRTUALCALL_LDVIRTFTN:
+                    if (!pResult->exactContextNeedsRuntimeLookup)
+                    {
+                        pResult->codePointerOrStubLookup.constLookup = CreateConstLookupToSymbol(_compilation.NodeFactory.DynamicHelperCell(
```

Comment:
We set the `constLookup` to different values above - can we avoid doing useless work in creating nodes we don't need? The nodes stay alive forever and get promoted from generation to generation.

Reply:
Okay I moved it up so we only make a single node. It looks less like ZapInfo::getCallInfo now but is the better way to go. To match CoreCLR we'll have to pull all the lookup handlers down into one place. Not sure if that would make things easier to understand tbh vs having the lookups / kinds done in one place above.